File: app/src/main/java/com/example/back2me/auth/SignupActivity.java
package com.example.back2me.auth;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.back2me.MainActivity;
import com.example.back2me.R;
import com.example.back2me.databinding.ActivitySignupBinding;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.UserProfileChangeRequest;

public class SignupActivity extends AppCompatActivity {

    private ActivitySignupBinding binding;
    private FirebaseAuth auth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySignupBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        auth = FirebaseAuth.getInstance();

        setupClickListeners();
    }

    private void setupClickListeners() {
        // Back button
        binding.backButton.setOnClickListener(v -> finish());

        // Sign up button
        binding.buttonSignup.setOnClickListener(v -> attemptSignup());

        // Login link
        binding.textLogin.setOnClickListener(v -> {
            finish(); // Go back to login
        });
    }

    private void attemptSignup() {
        String email = binding.inputEmail.getText().toString().trim();
        String password = binding.inputPassword.getText().toString().trim();
        String confirmPassword = binding.inputConfirmPassword.getText().toString().trim();

        // Validate
        if (TextUtils.isEmpty(email)) {
            binding.layoutEmail.setError("Email is required");
            return;
        }
        binding.layoutEmail.setError(null);

        if (TextUtils.isEmpty(password)) {
            binding.layoutPassword.setError("Password is required");
            return;
        }
        if (password.length() < 6) {
            binding.layoutPassword.setError("Password must be at least 6 characters");
            return;
        }
        binding.layoutPassword.setError(null);

        if (TextUtils.isEmpty(confirmPassword)) {
            binding.layoutConfirmPassword.setError("Please confirm your password");
            return;
        }
        if (!password.equals(confirmPassword)) {
            binding.layoutConfirmPassword.setError("Passwords don't match");
            return;
        }
        binding.layoutConfirmPassword.setError(null);

        // Show loading
        setLoading(true);

        // Create account
        auth.createUserWithEmailAndPassword(email, password)
                .addOnSuccessListener(authResult -> {
                    // Set display name from email
                    String displayName = email.split("@")[0];
                    UserProfileChangeRequest profileUpdates = new UserProfileChangeRequest.Builder()
                            .setDisplayName(displayName)
                            .build();

                    if (authResult.getUser() != null) {
                        authResult.getUser().updateProfile(profileUpdates)
                                .addOnCompleteListener(task -> {
                                    setLoading(false);
                                    Toast.makeText(this, "Account created successfully!", Toast.LENGTH_SHORT).show();

                                    Intent intent = new Intent(this, MainActivity.class);
                                    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                                    startActivity(intent);
                                    finish();
                                });
                    }
                })
                .addOnFailureListener(e -> {
                    setLoading(false);
                    Toast.makeText(this, "Sign up failed: " + e.getMessage(), Toast.LENGTH_LONG).show();
                });
    }

    private void setLoading(boolean loading) {
        binding.progressBar.setVisibility(loading ? View.VISIBLE : View.GONE);
        binding.buttonSignup.setEnabled(!loading);
        binding.buttonSignup.setText(loading ? "" : getString(R.string.sign_up));
    }
}

File: app/src/main/java/com/example/back2me/auth/SignupActivity.kt
// file: SignupActivity.kt (CORRECTED)
package com.example.back2me.auth

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.back2me.MainActivity
import com.example.back2me.databinding.ActivitySignupBinding
import com.google.firebase.auth.FirebaseAuth
import com.example.back2me.AuthData // ✅ KEEP: Needed for setUserId

// ❌ REMOVE: import java.util.UUID (Not needed)

class SignupActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySignupBinding
    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySignupBinding.inflate(layoutInflater)
        setContentView(binding.root)

        auth = FirebaseAuth.getInstance()

        setupClickListeners()
    }

    private fun setupClickListeners() {
        binding.backButton.setOnClickListener {
            finish()
        }

        binding.buttonSignup.setOnClickListener {
            val email = binding.inputEmail.text.toString().trim()
            val password = binding.inputPassword.text.toString().trim()
            val confirmPassword = binding.inputConfirmPassword.text.toString().trim()

            if (validateInput(email, password, confirmPassword)) {
                signupUser(email, password)
            }
        }

        binding.textLogin.setOnClickListener {
            finish()
        }

        // Social signup buttons (not implemented)
        binding.buttonGoogle.setOnClickListener {
            Toast.makeText(this, "Google signup not implemented", Toast.LENGTH_SHORT).show()
        }
        binding.buttonFacebook.setOnClickListener {
            Toast.makeText(this, "Facebook signup not implemented", Toast.LENGTH_SHORT).show()
        }
        binding.buttonTwitter.setOnClickListener {
            Toast.makeText(this, "Twitter signup not implemented", Toast.LENGTH_SHORT).show()
        }
    }

    private fun validateInput(email: String, password: String, confirmPassword: String): Boolean {
        if (email.isEmpty()) {
            binding.inputEmail.error = "Please enter your email"
            binding.inputEmail.requestFocus()
            return false
        }

        // Basic email validation
        if (!android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            binding.inputEmail.error = "Please enter a valid email"
            binding.inputEmail.requestFocus()
            return false
        }

        if (password.isEmpty()) {
            binding.inputPassword.error = "Please enter your password"
            binding.inputPassword.requestFocus()
            return false
        }

        if (password.length < 6) {
            binding.inputPassword.error = "Password must be at least 6 characters"
            binding.inputPassword.requestFocus()
            return false
        }

        if (confirmPassword.isEmpty()) {
            binding.inputConfirmPassword.error = "Please confirm your password"
            binding.inputConfirmPassword.requestFocus()
            return false
        }

        if (password != confirmPassword) {
            binding.inputConfirmPassword.error = "Passwords do not match"
            binding.inputConfirmPassword.requestFocus()
            return false
        }

        return true
    }

    private fun signupUser(email: String, password: String) {
        showLoading(true)

        auth.createUserWithEmailAndPassword(email, password)
            .addOnCompleteListener(this) { task ->
                showLoading(false)

                if (task.isSuccessful) {

                    // 💥 FIX: Get the Firebase User ID and save it.
                    val userId = task.result?.user?.uid
                    if (userId != null) {
                        AuthData.setUserId(userId)
                    }

                    Toast.makeText(this, "Account created successfully!", Toast.LENGTH_SHORT).show()
                    navigateToMain()
                } else {
                    val errorMessage = task.exception?.message ?: "Signup failed"
                    Toast.makeText(this, errorMessage, Toast.LENGTH_LONG).show()
                }
            }
    }

    private fun showLoading(show: Boolean) {
        binding.progressBar.visibility = if (show) View.VISIBLE else View.GONE
        binding.buttonSignup.isEnabled = !show
    }

    private fun navigateToMain() {
        val intent = Intent(this, MainActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        startActivity(intent)
        finish()
    }
}